### DISCUSSION TITLE
Custom Font

### DISCUSSION THREAD

**Discussion by aargon007** [tr3v3r/react-native-esc-pos-printer#165] [General]
Hi @tr3v3r, is there any option to use custom font family? the default font is awkward. thanks!

**Comment by tr3v3r** [comment]
@aargon007  hello there is a range of available fonts 

<img width="725" alt="image" src="https://github.com/user-attachments/assets/992ece12-c79c-4511-a2bd-995fc20c6736">


But to add the fully custom font the best way is generate image of receipt and then print that image 

.connect()
.addImage({ url: }) // can be generated on BE or using libs like View to Screenshot
.sendData()

**Comment by aargon007** [comment]
Hi @tr3v3r , can you help me where should i add this? 

` await printerInstance.addQueueTask(async () => {
                await Printer.tryToConnectUntil(
                    printerInstance,
                    status => status.online.statusCode === PrinterConstants.TRUE,
                );

                await printerInstance.addTextAlign(PrinterConstants.ALIGN_RIGHT);
                await printerInstance.addText('Final document of payment \n');
                await printerInstance.addText('Orii for flowershops \n');
                await printerInstance.addTextStyle({ em: PrinterConstants.TRUE } as const);
                await printerInstance.addText('www.orii.co');
                await printerInstance.addFeedLine(3);

                // Cut Paper
                await printerInstance.addCut();

                const result = await printerInstance.sendData();
                await printerInstance.disconnect();
}`

**Reply by tr3v3r** [reply]
@aargon007 to add snippet above? or what?

**Reply by aargon007** [reply]
@tr3v3r yeah, can you please give me a snippet? i am new and don't understand how to implement the SDK

**Reply by tr3v3r** [reply]
If you want to add this method use the following PR as an example. It adds addLineSpace method. You need to do the same for addTextFont 

https://github.com/tr3v3r/react-native-esc-pos-printer/pull/164/files

Original SDK doc is here https://github.com/tr3v3r/react-native-esc-pos-printer/blob/main/docs/SDK.md

**Comment by tr3v3r** [comment]
Just released a new version with an alternative way of printing.
Please check [here](https://github.com/tr3v3r/react-native-esc-pos-printer/blob/main/docs/printer/Printer.md#printeraddviewshotprinterinstance-printer-params-addviewshotparams-promisevoid)

**Reply by aargon007** [reply]
@tr3v3r thanks bro, it's fantastic.

**Comment by aargon007** [comment]
@tr3v3r add shot view perfectly work on android devices but i don't know why it's crashes on iOS devices

**Reply by tr3v3r** [reply]
please send me details

code and crash report please @aargon007

**Reply by aargon007** [reply]
i have try both on bare react native and expo project. it seems react-native-view-shot is not compattiple with expo. bare react native project work fine in both iOS and android. but expo app crash, i have upgraded react-native-view-shot to 4.0.0-alpha.3 and it not crashed anymore but printing blur

**Reply by tr3v3r** [reply]
@aargon007  try to add a background color to View. 

[<View /](https://stackoverflow.com/questions/74385095/expo-go-crashes-when-using-captureref-without-logging-anything)

**Reply by aargon007** [reply]
@tr3v3r now work perfectly.
![WhatsApp Image 2024-08-29 at 12 29 59 PM](https://github.com/user-attachments/assets/4260b95e-aaeb-4ddc-ace3-2ebb34aed013)

**Comment by aargon007** [comment]
closed
